### PR TITLE
Add URL as a possible input option to Vimeo player

### PR DESCRIPTION
**Pull request by yuriykuzin**
This is reasonable to processing correctly private Vimeo links, like https://vimeo.com/25345658/fe46e209ac - that contains not only vimeo id, but also a private key

### CODE REVIEW

File: src/js/plyr.js [hunk 1]
```diff
@@ -654,10 +654,10 @@
         return url.match(regex) ? RegExp.$2 : url;
     }
 
-    // Parse Vimeo ID from url
-    function _parseVimeoId(url) {
+    // Is this a Vimeo url link?
+    function _isVimeoUrl(url) {
```
Comment (stevermeister) [inline src/js/plyr.js]: it should not check and parse the same time, quite confusing

Comment (yuriykuzin) [reply src/js/plyr.js]: @stevermeister this function is only for checking, not parsing **anymore** ;)

Comment (stevermeister) [reply src/js/plyr.js]: true, sorry, I was comparing with `_parseVimeoId`

File: src/js/plyr.js [hunk 2]
```diff
@@ -1608,6 +1608,7 @@
         function _setupEmbed() {
             var container = document.createElement("div"),
                 mediaId,
+                mediaUrl,
```
Comment (stevermeister) [inline src/js/plyr.js]: you set it default to null

File: src/js/plyr.js [hunk 3]
```diff
@@ -1617,7 +1618,14 @@
...
 
                 case "vimeo":
-                    mediaId = _parseVimeoId(plyr.embedId);
+                    if (_isVimeoUrl(plyr.embedId)) {
+                        mediaId = null;
+                        mediaUrl = plyr.embedId;
+                    } else {
+                        mediaId = parseInt(plyr.embedId);
+                        mediaUrl = null;
+                    }
```
Comment (stevermeister) [inline src/js/plyr.js]: and set here only mediaUrl/mediaId

File: src/js/plyr.js [hunk 4]
```diff
@@ -1890,17 +1898,27 @@
         }
 
         // Vimeo ready
-        function _vimeoReady(mediaId, container) {
+        function _vimeoReady(mediaId, mediaUrl, container) {
```
Comment (stevermeister) [inline src/js/plyr.js]: let set options object here? 
that will extend vimeoOptions

Comment (stevermeister) [reply src/js/plyr.js]: or even better to keep vimeo structure:
`container, vimeoOptions`

**Pull request merged**